src/Symfony/Component/AssetMapper/ImportMap/ImportMapManager.php

```diff
@@ -168,15 +176,18 @@ private function buildImportMapJson(): void
         $dumpedModulePreloadPath = $this->assetsPathResolver->getPublicFilesystemPath().'/'.self::IMPORT_MAP_PRELOAD_FILE_NAME;
         if (is_file($dumpedImportMapPath) && is_file($dumpedModulePreloadPath)) {
             $this->json = file_get_contents($dumpedImportMapPath);
-            $this->modulesToPreload = json_decode(file_get_contents($dumpedModulePreloadPath), true, 512, \JSON_THROW_ON_ERROR);
+            $preload = json_decode(file_get_contents($dumpedModulePreloadPath), true, 512, \JSON_THROW_ON_ERROR);
+            $this->linkTags = $preload['linkTags'] ?? [];
+            $this->modulesToPreload = $preload['modules'] ?? [];
```

Comment:
Yea, good catch. With the current code, if you have the old format, you'll end up with an empty `modulesToPreload` property. So, basically, the old format will be ignored.

We could do something like `$this->modulesToPreload = $preload['modules'] ?? $preload` to catch the old format. But because the `assetmap:compile` command is meant to run on deploy, I don't think we need to. The user would upgrade `symfony/asset-mapper` locally. Then, when they deploy that, they'll re-run the compile command and get the new format immediately.

Reply:
this could cause issues if they ran the compile command locally.

Reply:
If you ran the compile command before upgrading, then upgraded, locally, the result is that the dumped file would effectively be ignored. That seems ok to me, right?

Reply:
don't we have a broken behavior if we load the compiled json without loading the compiled preload info ?

Reply:
In a minor way only: you'd just be missing the "module preloads". Everything would still work, you'd just have the performance hit of not having the preloads. And this, in practice, would only affect local dev (since the compile command should be run on deploy). That's why I'm thinking it's minor enough (+ the experimental on the component) to not introduce the code. But we can.